docs/decisions/0017-reimplement-asset-processing.rst

```diff
@@ -0,0 +1,268 @@
+
+       The overall asset *build* action will use the Bash script; this means that list of theme directories will need to be provided as arguments, but it ensures that the build can remain Python-free.
+   
+   * - **Collect** the built static assets from edx-platform to another location (the ``STATIC_ROOT``) so that they can be efficiently served *without* Django's webserver. This step, by nature, requires Python and Django in order to find and organize the assets, which may come from edx-platform itself or from its many installed Python and NPM packages. This is only needed for **production** environments, where it is usually desirable to serve assets with something efficient like NGINX.
+
+     - ``paver update_assets``
+
+       Paver task wrapping a call to the standard Django `collectstatic <https://docs.djangoproject.com/en/4.1/ref/contrib/staticfiles/#collectstati>`_ command. It adds ``--noinput`` and a list of ``--ignore`` file patterns to the command call.
+
+       (This command also builds assets. The *collect* action could not be run on its own without calling pavelib's Python interface.)
+
+     - ``./manage.py lms collectstatic --noinput && ./manage.py cms collectstatic --noinput``
+
+       The standard Django interface will be used without a wrapper. The ignore patterns will be added to edx-platform's `staticfiles app configuration <https://docs.djangoproject.com/en/4.1/ref/contrib/staticfiles/#customizing-the-ignored-pattern-list>`_ so that they do not need to be supplied as part of the command.
+   
+   * - **Watch** static assets for changes in the background. When a change occurs, rebuild them automatically, so that the Django webserver picks up the changes. This is only necessary in **development** environments. A few different sets of assets may be watched: XModule fragments, Webpack assets, default SCSS, and theme SCSS.
+
+     - ``paver watch_assets``
+
+       Paver task that invokes ``webpack --watch`` for Webpack assets and watchdog (a Python library) for other assets.
+
+     - ``assets/build.sh --watch <stage>``
+
+       (where ``<stage>`` is optionally one of the build stages described above. If provided, only that stage's assets will be watched.)
+
+       Bash wrappers around invocation(s) of `watchman <https://facebook.github.io/watchman/>`_, a popular file-watching library maintained by Meta. Watchman is already installed into edx-platform (and other services) via the pywatchman pip wrapper package.
+
+       Note: This adds a Python dependency to build.sh. However, we could be clear that watchman is an *optional* dependency of build.sh which enables the optional ``--watch`` feature. This would keep the *build* action Python-free. Alternatively, watchman is also available Python-free via apt and homebrew.
+
+Migration
+=========
+
+The old asset processing system will be `proposed for deprecation (TODO: link to issue) <http://example.com/depr>`_ upon provisional acceptance of this ADR.
```

Comment:
Another link insertion reminder; you could probably create the DEPR issue before merging, but only announce it afterwards.

Reply:
Will do.